Add codesage-large-v2

Aradhye2002 opened the pull request:
mteb/models/codesage_models.py

Fixes #2020 


<!-- If you are submitting a dataset or a model for the model registry please use the corresponding checklists below otherwise feel free to remove them. -->

<!-- add additional description, question etc. related to the new dataset -->


### Code Quality
<!-- Please do not delete this -->
- [ ] **Code Formatted**: Format the code using `make lint` to maintain consistent style.

### Testing
- [x] **Tests Passed**: Run tests locally using `make test` or `make test-with-coverage` to ensure no existing functionality is broken.


### Adding a model checklist
<!-- 
When adding a model to the model registry
see also https://github.com/embeddings-benchmark/mteb/blob/main/docs/reproducible_workflow.md
-->

 - [x] I have filled out the ModelMeta object to the extent possible
 - [x] I have ensured that my model can be loaded using
   - [x] `mteb.get_model(model_name, revision)` and
   - [x] `mteb.get_model_meta(model_name, revision)`
 - [ ] I have tested the implementation works on a representative set of tasks.

KennethEnevoldsen commented on mteb/models/codesage_models.py:
```diff
@@ -0,0 +1,113 @@
+    loader=partial(CodeSageModelWrapper, checkpoint="codesage/codesage-large-v2"),
+    name="codesage/codesage-large-v2",
+    languages=["code"],
+    revision=None,
```
required

KennethEnevoldsen commented on mteb/models/codesage_models.py:
```diff
@@ -0,0 +1,113 @@
+codesage_model_meta = ModelMeta(
+    loader=partial(CodeSageModelWrapper, checkpoint="codesage/codesage-large-v2"),
+    name="codesage/codesage-large-v2",
+    languages=["code"],
```
What about English?

KennethEnevoldsen commented on mteb/models/codesage_models.py:
```diff
@@ -0,0 +1,113 @@
+    revision=None,
+    release_date="2024-02-03",
+    modalities=["code"],
+    n_parameters=1300000000,
```
```suggestion
    n_parameters=1_300_000_000,
```

KennethEnevoldsen commented on mteb/models/codesage_models.py:
```diff
@@ -0,0 +1,113 @@
+    memory_usage_mb=None,
+    max_tokens=2048,
+    embed_dim=2048,
+    license=None,
```
Required

KennethEnevoldsen commented on mteb/models/codesage_models.py:
```diff
@@ -0,0 +1,113 @@
+    public_training_data=None,
+    framework=["PyTorch"],
+    reference="https://huggingface.co/codesage/codesage-large-v2",
+    similarity_fn_name="compute_similarity",
```
```suggestion
    similarity_fn_name="cosine",
```

KennethEnevoldsen reviewed (commented):
Seems like there are still a few things in the checklist missing notably the missing import of the model, means that it is not discoverable using `mteb.get_model(name)`

Aradhye2002 commented:
Thank you @KennethEnevoldsen, for the suggestions! I'll try my best to resolve the comments.

Samoed commented on mteb/models/codesage_models.py:
```diff
@@ -93,21 +93,21 @@ def compute_similarity(
-    languages=["code"],
-    revision=None,
+    languages=["eng_Latn"],
+    revision="2",
```
You should use revision from huggingface repo

Aradhye2002 replied:
Noted.

Samoed commented on mteb/models/codesage_models.py:
```diff
@@ -93,21 +93,21 @@ def compute_similarity(
     name="codesage/codesage-large-v2",
-    languages=["code"],
-    revision=None,
+    languages=["eng_Latn"],
```
This model supports more languages

Aradhye2002 replied:
Noted.

Samoed commented on mteb/models/codesage_models.py:
```diff
@@ -0,0 +1,113 @@
+    release_date="2024-02-03",
+    modalities=["text"],
+    n_parameters=1_300_000_000,
+    memory_usage_mb=None,
```
Can you calculate it using `memory_usage_mb`?

Aradhye2002 replied:
Let me try... Might be a bit tricky due to the large size of the model. I suppose I'll need to run it on GPUs?

Samoed commented on mteb/models/codesage_models.py:
```diff
@@ -0,0 +1,113 @@
+
+
+codesage_model_meta = ModelMeta(
+    loader=partial(CodeSageModelWrapper, checkpoint="codesage/codesage-large-v2"),
```
Why you didn't use sentence transformer?

Aradhye2002 replied:
Okay, I'll refactor the code to use sentence transformers.

Samoed replied:
I don't think that this is resolved. Why do you think we should add a new wrapper for the model?

Samoed commented on mteb/models/codesage_models.py:
```diff
@@ -0,0 +1,113 @@
+    reference="https://huggingface.co/codesage/codesage-large-v2",
+    similarity_fn_name="cosine",
+    use_instructions=False,
+    training_datasets=None,
```
Can you add datasets from paper?

Samoed replied:
Stack dataset, CodeSearchNet etc.

Samoed reviewed (commented):
Can you add small and medium versions too?

Aradhye2002 commented:
Hi @KennethEnevoldsen @Samoed. I've addressed all the comments so far and also made sure all tests pass locally (by running `make test`). Let me know if you want me to do anything else.

Samoed commented on mteb/models/codesage_models.py:
```diff
@@ -109,5 +109,7 @@ def compute_similarity(
     similarity_fn_name="cosine",
     use_instructions=False,
-    training_datasets=None,
+    training_datasets={
+        "TheStack.v2" : ["train"],
+    },
```
```suggestion
    training_datasets={
        # "TheStack.v2" : ["train"],
        "CodeSearchNetRetrieval": ["train"],
        "CodeSearchNetCCRetrieval": ["train"],
    },
```

Aradhye2002 replied:
Hi, I'm not sure if this is correct:

<img width="513" alt="Screenshot 2025-02-19 at 9 38 19 PM" src="https://github.com/user-attachments/assets/c43f7a5a-c217-48eb-bab0-2db96349e9bb" />
<img width="846" alt="Screenshot 2025-02-19 at 9 38 48 PM" src="https://github.com/user-attachments/assets/22d1a502-25cb-4f21-91dc-db8844013be9" />

Samoed replied:
<img width="567" alt="image" src="https://github.com/user-attachments/assets/1610214a-eb0f-4285-b50f-2081df80c61e" />

Aradhye2002 commented:
> Can you add small and medium versions too?

Done

Samoed commented:
Can you format your code?
